components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/IsMemberOfAnyOfGroupsFunctionImpl.java

```diff
@@ -66,6 +78,61 @@ public boolean isMemberOfAnyOfGroups(JsAuthenticatedUser user, List<String> grou
         return result;
     }
 
+    /**
+     * Check the federated user belongs to one of the groups specified in the list of user groups.
+     *
+     * @param user Authenticate user.
+     * @param groupNames      List of groups.
+     * @return whether the federated user belongs to one of the groups specified in the list of user groups .
+     */
+    private boolean isFederatedUserMemberOfAnyGroup(JsAuthenticatedUser user, List<String> groupNames) {
+
+        String[] groupsOfFederatedUser = null;
+        String groupsClaimValue = null;
+        String federatedIdPName = user.getWrapped().getFederatedIdPName();
+        String tenantDomain = user.getWrapped().getTenantDomain();
+        try {
+            // Get the claim URI for groups claim from the claim mappings for federated idp.
+            String groupsClaimURI = Utils.getGroupsClaimURIByClaimMappings(federatedIdPName, tenantDomain);
+            if (groupsClaimURI == null && user.getContext().getCurrentAuthenticator()
+                    .equals(OPENIDCONNECT_AUTHENTICATOR_NAME)) {
+                groupsClaimURI = DEFAULT_OIDC_GROUPS_CLAIM_URI;
+            }
+            if (groupsClaimURI == null) {
+                return false;
+            }
+            groupsClaimValue = getGroupsClaimValue(user, groupsClaimURI);
+            if (groupsClaimValue == null) {
+                return false;
+            }
+            groupsOfFederatedUser = groupsClaimValue.split(FrameworkUtils.getMultiAttributeSeparator());
+        } catch (IdentityProviderManagementException e) {
+            String msg = "Error while retrieving identity provider by name: " + federatedIdPName;
+            LOG.error(msg, e);
+        }
+        return Arrays.stream(groupsOfFederatedUser).anyMatch(groupNames::contains);
+    }
+
+    /**
+     * Get the value of the given groups claimURI.
+     *
+     * @param user Authenticate user.
+     * @param groupsClaimURI URI of the groups claim.
+     * @return groups claim value .
+     */
+    private String getGroupsClaimValue(JsAuthenticatedUser user, String groupsClaimURI) {
+
+        String groups = null;
+        if (StringUtils.isNotEmpty(groupsClaimURI) && MapUtils.isNotEmpty(user.getWrapped().getUserAttributes())) {
```

Comment:
Can the groupsClaimURI be empty at this point?

Reply:
It cannot be empty because it checks the claimURI is null before calling this method